src/quicksort/quicksort_recursive.nr

```diff
@@ -0,0 +1,51 @@
+pub trait Swap {
+    fn swap(&mut self, i: u32, j: u32);
+}
+
+impl<T, let N: u32> Swap for [T; N] {
+    fn swap(&mut self, i: u32, j: u32) {
+        let temp = self[i];
+        self[i] = self[j];
+        self[j] = temp;
+    }
+}
```

Comment:
I think we can remove this trait and just inline `swap` into`partition` as we don't use it anywhere else and it just adds another layer within the intepreter.